### ISSUE TITLE
@nx/vite:test no longer producing test result output

### ISSUE THREAD

**Issue by berenddeboer** [nrwl/nx#22223]
### Current Behavior

@nx/vite:test 17.1.3 used to produce output on what tests had been run. Typical output looked like this:

![image](https://github.com/nrwl/nx/assets/1514612/aa3a8baa-f880-48b7-89ed-cc0713be2ce7)

For some reason @nx/vite:test has not worked for us anymore (any version beyond 17.1.3).

The only thing you still see is the coverage report.

Another common result is seeing 1/1 failed, but no indication why:

![image](https://github.com/nrwl/nx/assets/1514612/50c22625-5680-484d-baa3-0908e7e07c51)

We can "fix" this by `pnpm i @nx/vite@17.1.3`in an otherwise 18.0.6 environment.

### Expected Behavior

18.0.6 should work as 17.1.3.

### Steps to Reproduce

1. Upgrade to any version after 17.1.3.
2. I've also experimented with copying packages/vite from the nx repo to my own monorepo. 17.1.3 works, copying anything produces no output.

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 18.19.0
OS     : darwin-arm64
pnpm   : 8.15.4

nx                 : 18.0.6
@nx/js             : 18.0.6
@nx/linter         : 18.0.6
@nx/eslint         : 18.0.6
@nx/workspace      : 18.0.6
@nx/cypress        : 18.0.6
@nx/devkit         : 18.0.6
@nx/esbuild        : 18.0.6
@nx/eslint-plugin  : 18.0.6
@nx/next           : 18.0.6
@nx/playwright     : 18.0.6
@nx/react          : 18.0.6
@nx/storybook      : 18.0.6
@nrwl/tao          : 18.0.6
@nx/vite           : 17.1.3
@nx/web            : 18.0.6
@nx/webpack        : 18.0.6
typescript         : 5.3.3
---------------------------------------
Community plugins:
@berenddeboer/nx-aws-cdk         : 2.2.2
@berenddeboer/nx-sst             : 0.13.5
@trumbitta/nx-plugin-unused-deps : 1.12.1
---------------------------------------
Local workspace plugins:
         @tc/nx-sst
         @tc/vite
---------------------------------------
The following packages should match the installed version of nx
  - @nx/vite@17.1.3
  - @nrwl/vite@17.1.3

To fix this, run `nx migrate nx@18.0.6`
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by berenddeboer** [comment]
Well, that was helpful, fixed it. I was missing the outputs option in the target!

**Comment by berenddeboer** [comment]
OK, I narrowed this down to the presence of a specific file. Even though that file is not even imported.

**Comment by berenddeboer** [comment]
Fixed by applying #20804 .
